src/lib/Libifl/pbsD_connect.c

```diff
@@ -594,6 +623,22 @@ __pbs_disconnect(int connect)
 
 	(void)destroy_connection(connect);
 
+	/* Update the server connection cache */
+	svr_conns = get_conn_servers();
+	if (svr_conns != NULL) {
+		int i;
+
+		for (i = 0; i < get_num_servers(); i++) {
+			if (svr_conns[i].sd == connect) {
+				svr_conns[i].sd = -1;
+				svr_conns[i].secondary_sd = -1;
```

Comment:
Shouldn't we need to close this and free its resources?

Reply:
good catch, ill make the change, thanks